### PR TITLE
erlcloud_ddb2: Add support for DynamoDB Global Table management

### PR DESCRIPTION
**Pull request by nalundgaard**
Adds functions to erlcloud_ddb2 to implement the following:

* CreateGlobalTable
* DescribeGlobalTable
* ListGlobalTables
* UpdateGlobalTable

### CODE REVIEW

File: src/erlcloud_ddb2.erl [hunk 1]
```diff
@@ -1871,6 +2027,69 @@ get_item(Table, Key, Opts, Config) ->
...
+     ]}.
+
+-type list_global_tables_return() :: ddb_return(#ddb2_list_global_tables{}, [#ddb2_global_table{}]).
+
+-spec list_global_tables() -> list_global_tables_return().
+list_global_tables() ->
+    list_global_tables([], default_config()).
+
+-spec list_global_tables(list_global_tables_opts()) -> list_global_tables_return().
+list_global_tables(Opts) ->
```
Comment (motobob) [inline src/erlcloud_ddb2.erl]: add a pedantic guard for `when is_list(Opts)`
as other apis allow `/1 (#aws_config{})`

File: src/erlcloud_ddb2.erl [hunk 2]
```diff
@@ -1644,6 +1753,53 @@ delete_table(Table, Opts, Config) ->
...
+     ]}. 
+
+-type describe_global_table_return() :: ddb_return(#ddb2_describe_global_table{}, #ddb2_global_table_description{}).
+
+-spec describe_global_table(table_name()) -> describe_table_return().
+describe_global_table(GlobalTableName) ->
+    describe_global_table(GlobalTableName, [], default_config()).
+
+-spec describe_global_table(table_name(), ddb_opts()) -> describe_table_return().
+describe_global_table(GlobalTableName, Opts) ->
```
Comment (motobob) [inline src/erlcloud_ddb2.erl]: guard for second argument here too.

File: src/erlcloud_ddb2.erl [hunk 3]
```diff
@@ -766,6 +776,29 @@ undynamize_expression(Expression, _) ->
...
+undynamize_replica_description(ReplicaDescription, _) ->
+    #ddb2_replica_description{region_name = proplists:get_value(<<"RegionName">>, ReplicaDescription)}.
+
+-spec undynamize_replica_descriptions(jsx:json_term(), undynamize_opts()) -> [replica_description()].
+undynamize_replica_descriptions(ReplicaDescriptions, Opts) ->
+    [undynamize_replica_description(ReplicaDescription, Opts)
+     || ReplicaDescription <- ReplicaDescriptions].
+
+-spec undynamize_replica(jsx:json_term(), undynamize_opts()) -> replica().
+undynamize_replica(Replica, _) ->
```
Comment (motobob) [inline src/erlcloud_ddb2.erl]: so sad these two c/p funs differ only in the actual record used...but this is outside of scope of the PR

File: src/erlcloud_ddb2.erl [hunk 4]
```diff
@@ -2356,6 +2575,66 @@ update_item(Table, Key, UpdatesOrExpression, Opts, Config) ->
...
+-type update_global_table_return() :: ddb_return(#ddb2_update_global_table{}, #ddb2_global_table_description{}).
+
+-spec update_global_table(table_name(), maybe_list(replica_update()))
+                         -> update_global_table_return().
+update_global_table(GlobalTableName, ReplicaUpdates) ->
+    update_global_table(GlobalTableName, ReplicaUpdates, [], default_config()).
+
+-spec update_global_table(table_name(), maybe_list(replica_update()), ddb_opts())
+                         -> update_global_table_return().
+update_global_table(GlobalTableName, ReplicaUpdates, Opts) ->
```
Comment (motobob) [inline src/erlcloud_ddb2.erl]: guard here?

File: src/erlcloud_ddb2.erl (hunk 3 above)
Comment (nalundgaard) [reply src/erlcloud_ddb2.erl]: Yeah, I decided to shadow their difference for [Replica](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Replica.html) and [ReplicaDescription](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ReplicaDescription.html). They are symmetric at this time, but it appears to me that they may change in the future, in particular, I think ReplicaDescription will gain additional fields once they, er... finish it?

File: src/erlcloud_ddb2.erl [hunk 5]
```diff
@@ -766,6 +777,29 @@ undynamize_expression(Expression, _) ->
...
+-spec undynamize_replica_descriptions(jsx:json_term(), undynamize_opts()) -> [replica_description()].
+undynamize_replica_descriptions(ReplicaDescriptions, Opts) ->
+    [undynamize_replica_description(ReplicaDescription, Opts)
+     || ReplicaDescription <- ReplicaDescriptions].
+
+-spec undynamize_replica(jsx:json_term(), undynamize_opts()) -> replica().
+undynamize_replica(Replica, _) ->
+    #ddb2_replica{region_name = proplists:get_value(<<"RegionName">>, Replica)}.
+
+-spec undynamize_replicas(jsx:json_term(), undynamize_opts()) -> [replica_description()].
```
Comment (paulo-ferraz-oliveira) [inline src/erlcloud_ddb2.erl]: Assuming `undynamize_replica/2` is -spec().ed OK, this function actually receives `([jsx:json_term()], _)` and outputs `[replica()]`.

**Pull request merged**